src/scripts.js

```diff
@@ -162,6 +162,49 @@ const getWeeklyOverview = () => {
 
   }
 
+const createFriendsSection = () => {
+  let friends = allUsers.userData.filter(user => {
+      return currentUser.userData.friends.includes(user.id) || user.id === indexOfUser
+    })
+  let friendsWeeklySteps = findFriendsWeeklySteps(friends)
+  var stepCounts =  findUserStepCounts(friendsWeeklySteps)
+
+  var sortedSteps = stepCounts.sort((firstValue,secondValue) => {
+  return secondValue.steps - firstValue.steps
+  })
+  addFriendsToHtml(sortedSteps)
+}
+findFriendsWeeklySteps = (friends) =>{
+  return friends.map(user => {
+    return { 
+      id : user.id,
+      date : activity.returnActvityWeeklyOverview(user.id,"2019/06/28",'numSteps')
+    }
+  })
+}
+addFriendsToHtml = (sortedSteps) =>{
+  sortedSteps.forEach(user =>{
+    var friendSection = document.createElement('div')
+    if(user.id === indexOfUser ){
+      friendSection.innerText = `You : ${user.steps} steps`
+
+    }else{
+    friendSection.innerText = `${allUsers.userData[user.id-1].name}: ${user.steps} steps`
+  }
+    friendLeaderBoard.appendChild(friendSection)
+  })
+}
+findUserStepCounts = (friendsWeeklySteps) => {
+  return friendsWeeklySteps.map(userSteps =>{
```

Comment:
It is hard to tell what is what here. Everything kind of chains together and there are no declarations of `const` or `let` before function names. It would also be beneficial to make sure that all variables are declared as let or const, instead of `var` to ensure that hoisting does not create an unintended behavior.